main/modes/games/swadgedoku/swadgedoku.c

```diff
@@ -0,0 +1,1370 @@
+
+    addSingleItemToMenu(sd->pauseMenu, menuItemResume);
+    addSingleItemToMenu(sd->pauseMenu, menuItemResetPuzzle);
+    addSingleItemToMenu(sd->pauseMenu, menuItemAbandonPuzzle);
+    addSingleItemToMenu(sd->pauseMenu, menuItemExitPuzzle);
+
+    swadgedokuSetupMenu();
+}
+
+static void swadgedokuExitMode(void)
+{
+    deinitSudokuGame(&sd->game);
+
+    void* val = NULL;
+    while (NULL != (val = pop(&sd->player.overlay.shapes)))
+    {
+        free(val);
+    }
+
+    freeFont(&sd->drawCtx.gridFont);
+    freeFont(&sd->drawCtx.noteFont);
+    freeFont(&sd->drawCtx.uiFont);
+
+    freeWsg(&sd->drawCtx.noteTakingIcon);
+
+    free(sd->player.notes);
+    free(sd->player.overlay.gridOpts);
+
+    deinitWheelMenu(sd->numberWheelRenderer);
+    deinitMenu(sd->numberWheel);
+
+    deinitMenuManiaRenderer(sd->menuRenderer);
+    deinitMenu(sd->menu);
+    deinitMenu(sd->emptyMenu);
+    deinitMenu(sd->pauseMenu);
+    free(sd);
+    sd = NULL;
+}
+
+static void swadgedokuMainLoop(int64_t elapsedUs)
+{
+    switch (sd->screen)
+    {
+        case SWADGEDOKU_MAIN_MENU:
+        {
+            buttonEvt_t evt = {0};
+            while (checkButtonQueueWrapper(&evt))
+            {
+                sd->menu = menuButton(sd->menu, evt);
+            }
+
+            drawMenuMania(sd->menu, sd->menuRenderer, elapsedUs);
+            break;
+        }
+
+        case SWADGEDOKU_GAME:
+        {
+            int32_t phi, r, intensity;
+            if (getTouchJoystick(&phi, &r, &intensity))
+            {
+                wheelMenuTouch(sd->numberWheel, sd->numberWheelRenderer, phi, r);
+                sd->touchState = true;
+            }
+            else if (sd->touchState)
+            {
+                wheelMenuTouchRelease(sd->numberWheel, sd->numberWheelRenderer);
+                sd->touchState = false;
+            }
+
+            // Now that touches have been handled, check if the wheel is active
+            bool inWheelMenu = wheelMenuActive(sd->numberWheel, sd->numberWheelRenderer);
+
+            if (!inWheelMenu)
+            {
+                sd->playTimer += elapsedUs;
+            }
+            // TODO I started to put an else here but I forgot what was supposed to go in it
+
+            buttonEvt_t evt = {0};
+            while (checkButtonQueueWrapper(&evt))
+            {
+                if (inWheelMenu)
+                {
+                    wheelMenuButton(sd->numberWheel, sd->numberWheelRenderer, &evt);
+                }
+                else
+                {
+                    swadgedokuGameButton(evt);
+                }
+            }
+
+            uint16_t* notes = sd->player.notes;
+            if (sd->settings.autoAnnotate)
+            {
+                notes = sd->game.notes;
+            }
+            swadgedokuDrawGame(&sd->game, notes, &sd->player.overlay, &lightTheme, &sd->drawCtx);
+
+            if (sd->player.noteTaking)
+            {
+                // draw like, a pencil
```

Comment:
I like the icon once I understood that pencil means notes. It's a little ambiguous at first b/c pencils write in solutions too. Maybe it'd be better as a notepad icon?